shell.nix

```diff
@@ -141,5 +151,6 @@ let
 in
 haskellNixShell // {
   cabalOnly = cabalShell;
+  demo = demoShell;
```

Comment:
These nix shells are so similar, I would be tempted to merge them for maintaining one nix expression less. Do you think the cost for the extra `hydra-cluster` in the demo and the extra `hydra-tui` in the ci scenario is too much?

Reply:
@yveshauser (he is not on `NixOS`) has experienced some problems with your git submodule spec (which uses `git` protocol). I thought that maybe it would be good to have a minimal setup without any extras. I don't know ;-)